shopwild-frontend/src/components/AllProducts/Product/Product.jsx

```diff
@@ -0,0 +1,169 @@
+import "./Product.css";
+
+import { useNavigate, Link } from "react-router-dom";
+
+import { useData } from "../../../context/data-context";
+
+import serverRequests from "../../../server/serverRequests";
+
+const Product = (props) => {
+  const { dispatch } = useData();
+  const navigate = useNavigate();
+
+  const utilToast = (setToast, setToastMessage, message) => {
+    setToast(true);
+    setToastMessage(message);
+    setTimeout(() => {
+      setToast(false);
+    }, 2000);
+  };
+
+  const clickHandler = async (e) => {
+    if (e.target.textContent === "ADD TO CART") {
+      const { error } = await serverRequests({
+        requestType: "post",
+        url: `${process.env.REACT_APP_BACKEND}/cart/607d92eee69f8b99745ef728`,
+        data: { products: [{ product: props.id, quantity: 1 }] },
```

Comment:
Instead of having a single clickHandler for both add to cart and add to wish and then again e.target.textContent in the click handler 
try creating seperate click handlers like addToCartHandler,addToWishHandler